scripts/globals/abilities/blitzers_roll.lua

```diff
@@ -48,12 +49,36 @@ function onUseAbility(caster,target,ability,action)
 end;
 
 function applyRoll(caster,target,ability,action,total)
-    local duration = 300 + caster:getMerit(MERIT_WINNING_STREAK)
+    local duration = 300 + caster:getMerit(MERIT_WINNING_STREAK) + caster:getMod(MOD_PHANTOM_DURATION)
     local effectpowers = {2, 3, 4, 11, 5, 6, 7, 8, 1, 10, 12, -5}
     local effectpower = effectpowers[total];
---    if (caster:getLocalVar("corsairRollBonus") == 1 and total < 12) then -- TODO Add AF3
---        effectpower = effectpower + 15 --  TODO Add Logic for Phantom Roll+
---    end
+-- Apply Buffs from Blitzers Roll Enhancing Gear if present
+    local equippedEnh = caster:getEquipID(SLOT_HEAD);
+    if (equippedEnh == 11180) then -- Navarch's Tricorne +1
+        if (math.random(0, 99) < 50) then  -- 50% Chance to buff roll if equipped
+            effectpower = effectpower + 3
+        end
+    end
+    if ((equippedEnh == 11080) or (equippedEnh == 26772) or (equippedEnh == 26773)) then -- Navarch's Tricorne +2 / Chasseur's Tricorne / +1
+        effectpower = effectpower + 3
+    end
+-- Check for MOD_PHANTOM_ROLL Value and apply non-stack logic.
+    local phantomValue = caster:getMod(MOD_PHANTOM_ROLL);
```

Comment:
I see this block in a lot of places, maybe you can just put it in its own function in ability.lua?

Reply:
Yep, that's definitely doable. I'll relocate that function to ability.lua and call it instead.